pkg/kubevirt/core/core_test.go

```diff
@@ -0,0 +1,162 @@
+package core
+
+import (
+	"context"
+	api "github.com/gardener/machine-controller-manager-provider-kubevirt/pkg/kubevirt/apis"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
```

Comment:
Why should I reinvent the wheel :-). This client is giving the tests all what it needs.

Reply:
This is not about reinventing the wheel, it's about a more comprehensive / universal approach to testing. Interface mocks are universal, can be generated for any interface (not just `client.Client`), and can be easily setup to imitate any behavior. If a mock is called with the wrong number / values of parameters, it's immediately an error. All this you don't get with the `fake` package. 
Note that this is also the approach used in much of the Gardener codebase, it should be used here as well. This can be done in a new PR as I mentioned above.

Reply:
Well, again, this universal approach can be valid of course if you have a type that needs to be mocked. But here you have one which is the fake client! a universal approach for dealing with k8s objects when testing. Another approach could be using ClientSet with Invokes however the case that I am testing doesn't need that layer of complexity. I am just creating, getting and deleting objects. If I need to react on something I can add da reactors in there, but this part of the rest client more than the controller runtime one! Otherwise mocking these clients is just a boilerplate as there is no special thing to do here!

Reply:
Anyway, if your codebase is assuming that each client has to be mocked explicitly then sure we can do that.

Reply:
So as agreed during the meeting sync, will update the test file in a separate PR.